docs/reference/contributing/guidelines/workflow.md

```diff
@@ -139,6 +141,20 @@ A breaking change is any change that results in breaking user space. It should h
 
 A contribution containing a breaking change is the most difficult PR to get merged. Any breaking changes in a codebase can have a large negative impact on any users of the codebase. Breaking changes are always limited to a major version release.
 
+Every pull request with breaking change must contain a release notes section called "Release notes" to describe the changes to users.
+
+It must contain:
+
+- A brief description of changes introduced, including justification description.
+- An analysis of effects: components affected, potential consequences for users and reasons for breaking user space.
+- Migration guidance: actions for updating the current code. Please include code snippets to illustrate before and after the addition or change.
```

Comment:
Where do the code snippets go? Will these end up in the release notes for, for example, Mbed OS 5.16; or perhaps a whole Mbed OS 5.15 to Mbed OS 5.16 migration guide?

Reply:
At the moment will be part of release notes. We do not have "migration guide" concept yet.